test/compiler.test.js

```diff
@@ -405,6 +405,17 @@ describe('compiler', function() {
       expect(instructions.files.boot).to.eql([initJs]);
     });
 
+    it('should resolve js files in `bootScripts`', function() {
```

Comment:
The test name IMO does not describe the scenario well. The only relevant distinction of this test is that the bootScript file does not provide extension, right? My proposal: "resolves missing file extensions in `bootScritps`", but feel free to come up with your own name.

In the modules I start from scratch, I prefer to use the present tense in test names instead of starting each test name with `should` - [DRY](http://en.wikipedia.org/wiki/Don%27t_repeat_yourself). (Even though older loopback modules use the "should" convention.) Please keep the test names consistent with the convention used in each project.